crates/jit/src/profiling/vtune_linux.rs

```diff
@@ -32,15 +33,21 @@ pub struct VTuneAgent {
     state: Mutex<State>,
 }
 
+#[derive(Clone, Debug, Hash, PartialEq, Eq)]
+enum MethodKind {
+    Func { module_id: usize, func_id: usize },
+    ExportedFuncTrampoline { module_id: usize, func_id: usize },
+}
+
 /// Interface for driving vtune
 #[derive(Clone, Debug, Default)]
 struct State {
-    /// Unique identifier for the jitted function
-    method_id: HashMap<(usize, DefinedFuncIndex), u32>,
+    /// Unique identifier for the jitted code region.
+    method_id: HashMap<MethodKind, u32>,
```

Comment:
Reading over this now, there's actually a fair bit supporting this map right here, but this map is only used for an assert that each item inserted is unique. I think that may mean perhaps that we could remove this entirely? (and not worry about `MethodKind` as a new enum)

Reply:
Indeed, the module id we feed ourselves and the signature index will always be different anyways, so this sanity check isn't worth the trouble.

Kept the mutex, though, because I'm unclear whether the ittapi function can be called from several threads at the same time :thinking: